### DISCUSSION TITLE
How can I get a session variable or any ways parameters before run a report in web portal, and use this variable to filter in SQL condition?

### DISCUSSION THREAD

**Discussion by aazone123** [ariacom/Seal-Report#124] [General]
Any way to get the request parameter (e.g. session variable, POST/GET URL parameter, HTML form elements ? )
Is it is can be handle by using "Additional WHERE clause" ? and write the Razor scripts (C#) to handle it?

Anyone can help and advise?
Thanks.

**Comment by ariacom** [comment]
If run from the Web Server, the report has a SecurityContext (class SecurityUser) set: 
        var securityContext = report.SecurityContext; //User is logged through a Web Report Server and has a context
        restriction = securityContext.Name; //Name of the user set during the login
        restriction = securityContext.WebUserName; //Name got from the login window 
        //securityContext.SecurityGroups; //List of security groups set for the user
        if (securityContext.BelongsToGroup("Default Group")) { //Test if the user belongs to a group
            //Special restriction here
        }

During the login script, you can set use its Tag property for additional information you can use later.

**Reply by aazone123** [reply]
Hi ariacom,

I have HTML javascript code below, which is a webpage to call a seal report. And defined this variable --> OrderNo: "myOrderNo123".
Please help, How can I retrieve this variable from my seal report? So that I can apply this varaible "myOrderNo123" to SQL query?
What I need to do in my seal report designer?
Thanks.

        var data = {
            path: "\\testReport1.srex",
            sessionId: glbSessionId,
            OrderNo: "myOrderNo123",
            use_default_restrictions: "true"
        };

        var form = document.createElement("form");
        form.method = "POST";
        form.action = sealServer + "SWExecuteReport";
        form.target = "_blank";

        for (var key in data) {
            if (data.hasOwnProperty(key)) {
                var hiddenField = document.createElement("input");
                hiddenField.type = "hidden";
                hiddenField.name = key;
                hiddenField.value = data[key];
                form.appendChild(hiddenField);
            }
        }

        document.body.appendChild(form);
        form.submit();

**Reply by ariacom** [reply]
if you want to pass information during the report execution, it should be one of the Restriction/Input Value of the report,
it will be the same as described in https://sealreport.org/demo/WebInterfaceAPI.html
Thus you have to create the restriction in the report first.
